test/src/tests-identities-attributes.js

```diff
@@ -1163,4 +1163,46 @@ describe('identities and attributes', function() {
 
         done();
     });
+
+    it('should not send user attribute change requests when user attribute already set with same value', function(done) {
```

Comment:
This test would pass even with the original code for `if (!previousUserAttributeValue)`.  Please add tests for where the attribute is `0`, `false`, and `undefined`.

Reply:
@rmi22186 should I also remove the test case for attribute age: 25?

Reply:
i think you can add a few user attributes, so it's fine to keep them and just add the others.